### PR TITLE
Add archive diagnostic files when compile fails

### LINKED ISSUE #6331: Capture diagnostic files created during build job
In https://github.com/eclipse/openj9/issues/6330 we got a crash during the build process. Ideally the build job should capture the core file and archive it for diagnostics.

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Catch compile failures and tar any diagnostic files
- Diagnostic files include
   - core.*.dmp
   - javacore.*.txt
   - Snap.*.trc
   - jitdump.*.dmp
- Upload to Artifactory if set otherwise archive to
  Jenkins master.

Fixes #6331
Depends #6346 
[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
Appears to be working now.
https://140-211-168-230-openstack.osuosl.org/artifactory/webapp/#/artifacts/browse/tree/General/ci-eclipse-openj9/Build_JDK11_ppc64le_linux_Personal/224/Build_JDK11_ppc64le_linux_Personal-224-20190715-141523-diagnostics.tar.gz
```
00:27:38.987  Compiling 4 files for BUILD_JIGSAW_TOOLS
00:27:38.987  Unhandled exception
00:27:38.987  Type=Segmentation error vmState=0xffffffff
00:27:38.987  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
00:27:38.987  Handler1=000010000099DCF0 Handler2=0000100000B3E170
00:27:38.987  R0=00001000009BA4FC R1=000010000085C880 R2=0000100000ABBF00 R3=000010000400F8C0
00:27:38.987  R4=0000000000000000 R5=FFFFFFFFFFFFFFFF R6=0000000000000000 R7=0000000000000000
00:27:38.987  R8=0000000000000074 R9=0000000000000000 R10=000000000000006F R11=0000100001091410
00:27:38.987  R12=0000000024004422 R13=0000100000866900 R14=00000100212C1650 R15=00000100212C18F0
00:27:38.987  R16=000010000085D180 R17=0000000000000000 R18=00001000008AA300 R19=0000100000A929B0
00:27:38.987  R20=000010000085CCB8 R21=0000100000AA6D68 R22=000010000400D0E0 R23=0000100000A904A8
00:27:38.987  R24=000010000400D250 R25=000010000400D0E0 R26=0000100000AB0C98 R27=0000000000000000
00:27:38.987  R28=000010000401E538 R29=0000000000000000 R30=0000000000000000 R31=000010000400F8C0
00:27:38.987  NIP=00001000009ED1F4 MSR=800000010280F033 ORIG_GPR3=0000100002108D88 CTR=000010000021F080
00:27:38.987  LINK=00001000009BA550 XER=0000000020000000 CCR=0000000084004428 SOFTE=0000000000000001
00:27:38.987  TRAP=0000000000000300 DAR=0000000000000000 dsisr=0000000042000000 RESULT=0000000000000000
00:27:38.987  FPR0 00001000008c3468 (f: 9188456.000000, d: 8.691699e-311)
00:27:38.987  FPR1 4130624de0000000 (f: 3758096384.000000, d: 1.073742e+06)
00:27:38.987  FPR2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR6 6b646a2f65736165 (f: 1702060416.000000, d: 2.097354e+209)
00:27:38.987  FPR7 65732d6c616d726f (f: 1634562688.000000, d: 4.973567e+180)
00:27:38.987  FPR8 366370705f31314b (f: 1597059456.000000, d: 1.064068e-46)
00:27:38.987  FPR9 6e6f737265505f78 (f: 1699766144.000000, d: 9.094959e+223)
00:27:38.987  FPR10 4000000000000000 (f: 0.000000, d: 2.000000e+00)
00:27:38.987  FPR11 3fe0000000000000 (f: 0.000000, d: 5.000000e-01)
00:27:38.987  FPR12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:27:38.987  Module=/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/build/linux-ppc64le-normal-server-release/jdk/lib/compressedrefs/libj9vm29.so
00:27:38.987  Module_base_address=00001000008F0000
00:27:38.987  Target=2_90_20190715_224 (Linux 4.4.0-141-generic)
00:27:38.987  CPU=ppc64le (16 logical CPUs) (0x1fe0b0000 RAM)
00:27:38.987  JVMDUMP039I Processing dump event "gpf", detail "" at 2019/07/15 10:42:25 - please wait.
00:27:38.987  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/core.20190715.104225.3976.0001.dmp' in response to an event
00:27:38.987  JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.3979.
00:27:38.987  
00:27:38.988  JVMDUMP010I System dump written to /home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/core.20190715.104225.3976.0001.dmp
00:27:38.988  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/javacore.20190715.104225.3976.0002.txt' in response to an event
00:27:38.988  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/Snap.20190715.104225.3976.0003.trc' in response to an event
00:27:38.988  JVMDUMP009I Snap dump not available
00:27:38.988  JVMDUMP013I Processed dump event "gpf", detail "".
00:27:38.988  gmake[3]: *** [/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/build/linux-ppc64le-normal-server-release/jdk/_packages_attribute.done] Error 255
00:27:38.988  ExplodedImageOptimize.gmk:40: recipe for target '/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/build/linux-ppc64le-normal-server-release/jdk/_packages_attribute.done' failed
00:27:38.988  make/Main.gmk:347: recipe for target 'exploded-image-optimize' failed
00:27:38.988  gmake[2]: *** [exploded-image-optimize] Error 1
00:27:38.988  gmake[2]: *** Waiting for unfinished jobs....
00:27:39.963  
00:27:39.963  ERROR: Build failed for target 'all' in configuration 'linux-ppc64le-normal-server-release' (exit code 2) 
00:27:39.963  Stopping sjavac server
00:27:39.963  
00:27:39.963  === Make failed targets repeated here ===
00:27:39.963  ExplodedImageOptimize.gmk:40: recipe for target '/home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/build/linux-ppc64le-normal-server-release/jdk/_packages_attribute.done' failed
00:27:39.963  make/Main.gmk:347: recipe for target 'exploded-image-optimize' failed
00:27:39.963  === End of repeated output ===
00:27:39.963  
00:27:39.963  Hint: Try searching the build log for the name of the first failed target.
00:27:39.963  Hint: See doc/building.html#troubleshooting for assistance.
00:27:39.963  
00:27:39.963  /home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/Init.gmk:300: recipe for target 'main' failed
00:27:39.963  make[1]: *** [main] Error 1
00:27:39.963  /home/jenkins/workspace/Build_JDK11_ppc64le_linux_Personal/make/Init.gmk:186: recipe for target 'all' failed
00:27:39.963  make: *** [all] Error 2
[Pipeline] sh
00:27:43.674  + + find . -nametar core.*.dmp -zcvf -o Build_JDK11_ppc64le_linux_Personal-224-20190715-141523-diagnostics.tar.gz -name -T javacore.*.txt - -o
00:27:43.674   -name Snap.*.trc -o -name jitdump.*.dmp
00:27:43.674  ./make/javacore.20190715.104225.3976.0002.txt
00:27:43.674  ./make/core.20190715.104225.3976.0001.dmp
[Pipeline] getArtifactoryServer
[Pipeline] newBuildInfo
[Pipeline] artifactoryUpload
00:27:46.993  [consumer_0] Deploying artifact: https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDK11_ppc64le_linux_Personal/224/Build_JDK11_ppc64le_linux_Personal-224-20190715-141523-diagnostics.tar.gz
[Pipeline] publishBuildInfo
00:28:05.400  Deploying build info to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
00:28:05.796  Deploying build descriptor to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
00:28:05.991  Build successfully deployed. Browse it in Artifactory under https://140-211-168-230-openstack.osuosl.org/artifactory/webapp/builds/Build_JDK11_ppc64le_linux_Personal/224
00:28:05.991  Sending request for build retention, deleting build artifacts, max number of builds to store: 30.
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // stage
[Pipeline] cleanWs
00:28:07.890  [WS-CLEANUP] Deleting project workspace...
00:28:07.890  [WS-CLEANUP] Deferred wipeout is disabled by the job configuration...
00:28:55.542  [WS-CLEANUP] done
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // timeout
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE
```

**Comment by AdamBrousseau** [comment]
@vsebe for review.
I will leave as Draft PR until we are ready to merge since I have a code change in there which throws the seg fault.

**Comment by AdamBrousseau** [comment]
jenkins compile win jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile plinx jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile plinux jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk11

**Comment by AdamBrousseau** [comment]
Commands work on Linux, AIX and Windows.

File: runtime/vm/vmthread.c [hunk 1]
```diff
@@ -497,8 +497,10 @@ threadParseArguments(J9JavaVM *vm, char *optArg)
 #endif /* OMR_THR_ADAPTIVE_SPIN */
 #endif /* OMR_THR_CUSTOM_SPIN_OPTIONS */
 #endif /* J9VM_INTERP_CUSTOM_SPIN_OPTIONS */
-	PORT_ACCESS_FROM_JAVAVM(vm);
+	int *p = NULL;
```
Comment (vsebe) [inline runtime/vm/vmthread.c]: please revert this change

**Comment by AdamBrousseau** [comment]
@pshipton for merge
